src/lib/clk.c

```diff
@@ -112,11 +112,11 @@ void clock_set_freq(int clock, uint32_t hz)
 		notify_data.target_core_mask =
 			NOTIFIER_TARGET_CORE_MASK(cpu_get_id());
 		break;
-	case CLK_SSP:
-		set_freq = &clock_platform_set_ssp_freq;
-		freq_table = ssp_freq;
-		freq_table_size = ARRAY_SIZE(ssp_freq);
-		notify_data.id = NOTIFIER_ID_SSP_FREQ;
+	case CLK_I2S:
+		set_freq = &clock_platform_set_i2s_freq;
```

Comment:
I am not sure I see the point of using I2S which is a subset of the protocols supported by most serial interfaces.

Rather than changing all the code, maybe this should be abstracted with a pointer in platform-specific ops?

Reply:
@jgunn0262 I'm assuming this is for Hikey960 ? I agree SSP is too Intel specific for core clk API.
@plbossart agreed, it needs some abstraction or conditional compilation.
@dbaluta fyi, I guess you have the same issue, how are you solving ?

Reply:
@lgirdwood yes it's for Hikey960, let me see if I can abstract it more.

Reply:
@jgunn0262 do you mean this one: https://www.amazon.com/dp/B07CGGC9NB/ref=dp_prsubs_2

Reply:
@plbossart yes, that's the board.